### PR TITLE
Add coinbase support

### PR DESCRIPTION
**Pull request by buddylindsey**
This is to start issue #466 It is basic for now to get something people can build off of, and to get started if there are any crucial base problems that need to be resolved.

I feel like this was a wild guess on getting it right. I looked at several exchanges files as well as the docs and I feel like this is a good start, but it is probable I missed something. Or I misunderstood something.

The main problem so far is how do I upload a coinbase image to github for use in this?

### CODE REVIEW

**Comment by kroitor** [comment]
Some endpoint are missing there, probably...

Here's the full list, which is also somewhat outdated, likely:

```
GET https://api.coinbase.com/v2/users/:user_id
GET https://api.coinbase.com/v2/user
GET https://api.coinbase.com/v2/user/auth
PUT https://api.coinbase.com/v2/user
GET https://api.coinbase.com/v2/accounts
GET https://api.coinbase.com/v2/accounts/:account_id
POST https://api.coinbase.com/v2/accounts
POST https://api.coinbase.com/v2/accounts/:account_id/primary
PUT https://api.coinbase.com/v2/accounts/:account_id
DELETE https://api.coinbase.com/v2/accounts/:account_id
GET https://api.coinbase.com/v2/accounts/:account_id/addresses
GET https://api.coinbase.com/v2/accounts/:account_id/addresses/:address_id
GET https://api.coinbase.com/v2/accounts/:account_id/addresses/:address_id/transactions
POST https://api.coinbase.com/v2/accounts/:account_id/addresses
GET https://api.coinbase.com/v2/accounts/:account_id/transactions
GET https://api.coinbase.com/v2/accounts/:account_id/transactions/:transaction_id
POST https://api.coinbase.com/v2/accounts/:account_id/transactions
POST https://api.coinbase.com/v2/accounts/:account_id/transactions
POST https://api.coinbase.com/v2/accounts/:account_id/transactions
POST https://api.coinbase.com/v2/accounts/:account_id/transactions/:transaction_id/complete
POST https://api.coinbase.com/v2/accounts/:account_id/transactions/:transaction_id/resend
DELETE https://api.coinbase.com/v2/accounts/:account_id/transactions/:transaction_id
GET https://api.coinbase.com/v2/accounts/:account_id/buys
GET https://api.coinbase.com/v2/accounts/:account_id/buys/:buy_id
POST https://api.coinbase.com/v2/accounts/:account_id/buys
POST https://api.coinbase.com/v2/accounts/:account_id/buys/:buy_id/commit
GET https://api.coinbase.com/v2/accounts/:account_id/sells
GET https://api.coinbase.com/v2/accounts/:account_id/sells/:sell_id
POST https://api.coinbase.com/v2/accounts/:account_id/sells
POST https://api.coinbase.com/v2/accounts/:account_id/sells/:sell_id/commit
GET https://api.coinbase.com/v2/accounts/:account_id/deposits
GET https://api.coinbase.com/v2/accounts/:account_id/deposits/:deposit_id
POST https://api.coinbase.com/v2/accounts/:account_id/deposits
POST https://api.coinbase.com/v2/accounts/:account_id/deposits/:deposit_id/commit
GET https://api.coinbase.com/v2/accounts/:account_id/withdrawals
GET https://api.coinbase.com/v2/accounts/:account_id/withdrawals/:withdrawal_id
POST https://api.coinbase.com/v2/accounts/:account_id/withdrawals
POST https://api.coinbase.com/v2/accounts/:account_id/withdrawals/:withdrawal_id/commit
GET https://api.coinbase.com/v2/payment-methods
GET https://api.coinbase.com/v2/payment-methods/:payment_method_id/
GET https://api.coinbase.com/v2/merchants/:merchant_id
GET https://api.coinbase.com/v2/orders
GET https://api.coinbase.com/v2/orders/:order_id
POST https://api.coinbase.com/v2/orders
POST https://api.coinbase.com/v2/orders/:order_id/refund
GET https://api.coinbase.com/v2/checkouts
GET https://api.coinbase.com/v2/checkouts/:checkout_id
POST https://api.coinbase.com/v2/checkouts
GET https://api.coinbase.com/v2/checkouts/:checkout_id/orders
POST https://api.coinbase.com/v2/checkouts/:checkout_id/orders
GET https://api.coinbase.com/v2/currencies
GET https://api.coinbase.com/v2/exchange-rates
GET https://api.coinbase.com/v2/prices/:currency_pair/buy
GET https://api.coinbase.com/v2/prices/:currency_pair/sell
GET https://api.coinbase.com/v2/prices/:currency_pair/spot
GET https://api.coinbase.com/v2/time
mbp:ccxt igorkroitor$
```

File: ccxt.js [hunk 1]
```diff
@@ -84,6 +84,7 @@ const exchanges = {
     'chbtc':                   require ('./js/chbtc.js'),
     'chilebit':                require ('./js/chilebit.js'),
     'cobinhood':               require ('./js/cobinhood.js'),
+    'coinbase':                require ('./js/coinbase.js'),
```
[the remark this answers is not in the document]
Comment (buddylindsey) [reply ccxt.js]: Thanks. I missed that. 👍

**Comment by kroitor** [comment]
I'll get this merged so that others could start building on top... Will need some time to make sure everything is correct and for some testing...

**Comment by buddylindsey** [comment]
How did you get that list? I'll make sure none are later today/tomorrow. I mostly likely missed some because I was scrolling through the docs and adding them, heh.

**Comment by kroitor** [comment]
> How did you get that list? 

Well, that is usually a painful manual process ) Just stripping them off of their docs... If they had a swagger description or something this could be automated to some extent... But anyways, my list can be outdated as well. I'll check for that and will add the missing endpoints along the way, no worries. )

**Comment by kroitor** [comment]
> The main problem so far is how do I upload a coinbase image to github for use in this?

I usually add it manually, creating it in an image editor, you don't have to worry about the image, as I'll create one upon merging. )

**Pull request merged**